ddg_api.py

```diff
@@ -7,27 +7,48 @@
 cors = CORS(app, resources={r"/*":
                             {"origins": "http://localhost:3000"}})
 
+global saved_mols
+# Can't serialize sets using json
+saved_mols = []
+
 
 @app.route("/")
 def hello_world():
     return "<p>Hello, World!</p>"
 
 
+# Pass molecule IDs as queries: /save?r1=A01&r2=B10
+@app.route("/save", methods=['POST'])
+def save_molecule():
+    r_group_1_id = request.args.get('r1')
+    r_group_2_id = request.args.get('r2')
+    if (r_group_1_id, r_group_2_id) not in saved_mols:
+        saved_mols.append((r_group_1_id, r_group_2_id))
+    print(saved_mols)
+    return jsonify({'saved_mols': saved_mols})
+
+
+@app.route("/savedmolecules")
+def return_saved_molecules():
+    return jsonify({'saved_mols': saved_mols})
+
+
 @app.route("/r-group-<string:r_group_id>")
 def rgroup_img(r_group_id):
-    """Returns image of R group specified by ID as a
-    bytestream to be rendered in a browser.
+    """Returns image and stats R group specified by ID as a
+    bytestream and dict to be rendered in a browser.
 
     :param r_group_id: ID number of R Group, eg. 'B26'
     :type r_group_id: String
-    :return: Image of R Group as bytstream in a json dict.
-    Access image bytestream with `img_html` key
+    :return: Image and stats of R Group in a json dict.
+    Access image bytestream with `img_html` key and stats with 'stats'
     :rtype: json dict
     """
-    smile_string = get_smile_string(r_group_id)
-    mol = Molecule(smile_string)
+    # smile_string = get_smile_string(r_group_id)
+    # mol = Molecule(smile_string)
+    mol = R_group(r_group_id)
     bytestream = mol.drawMoleculeAsByteStream()
-    stats_dict = (mol.descriptors())
+    stats_dict = mol.descriptors()
     return jsonify({'img_html': f"data:;base64,{bytestream}",
```

Comment:
I don't understand 

f"data:;base64,{bytestream}

exactly - is the syntax OK?

Reply:
Also, where is it getting img_html from?

Reply:
its a python way of string formatting. Its equivalent to 'data:;base64, + bytstream in this case. The data:;base64 is the file header for the webbrowser, it tells it how to parse the bytestream to geenrate the image